test/browser/anchor-manager.test.jsx

```diff
@@ -0,0 +1,261 @@
+      'Anchor mismatch for #foo.'
+    ]);
+    console.warn.restore();
+  });
+
+  it('should no-op if hash changes between unmatched anchors', function() {
+    var anchor = manager.register(fakeAnchor('foo'));
+    manager.simulateHashChange('blah');
+    anchor.handleNavigateFromAnchor.callCount.should.equal(0);
+    anchor.handleNavigateToAnchor.callCount.should.equal(0);
+    should(manager.currentAnchor).be.null;
+  });
+
+  it('should no-op if hash matches current anchor', function() {
+    var anchor = manager.register(fakeAnchor('foo'));
+    manager.currentAnchor = anchor;
+    manager.simulateHashChange('foo');
+    anchor.handleNavigateFromAnchor.callCount.should.equal(0);
+    anchor.handleNavigateToAnchor.callCount.should.equal(0);
+    manager.currentAnchor.should.equal(anchor);
+  });
+
+  it('should navigate from unmatched anchor to matched anchor', function() {
+    var anchor = manager.register(fakeAnchor('foo'));
+    manager.simulateHashChange('foo');
+    anchor.handleNavigateFromAnchor.callCount.should.equal(0);
+    anchor.handleNavigateToAnchor.callCount.should.equal(1);
+    manager.currentAnchor.should.equal(anchor);
+  });
+
+  it('should navigate from old anchor to unmatched anchor', function() {
+    var fooAnchor = manager.register(fakeAnchor('foo'));
+    manager.currentAnchor = fooAnchor;
+    manager.simulateHashChange('blah');
+    fooAnchor.handleNavigateFromAnchor.callCount.should.equal(1);
+    fooAnchor.handleNavigateToAnchor.callCount.should.equal(0);
+    should(manager.currentAnchor).be.null;
+  });
+
+  it('should navigate from old anchor to matched anchor', function() {
+    var fooAnchor = manager.register(fakeAnchor('foo'));
+    var barAnchor = manager.register(fakeAnchor('bar'));
+    manager.currentAnchor = fooAnchor;
+    manager.simulateHashChange('bar');
+    fooAnchor.handleNavigateFromAnchor.callCount.should.equal(1);
+    fooAnchor.handleNavigateToAnchor.callCount.should.equal(0);
+    barAnchor.handleNavigateFromAnchor.callCount.should.equal(0);
+    barAnchor.handleNavigateToAnchor.callCount.should.equal(1);
+    manager.currentAnchor.should.equal(barAnchor);
+  });
+});
+
+describe('AnchorManagerMixin', function() {
+  var manager, anchors, clock;
+
+  var SampleAnchorClass = React.createClass({
+    mixins: [AnchorManagerMixin],
+    render: function() {
+      return <div></div>;
+    }
+  });
+
+  var renderAnchor = function(props) {
+    var anchor = React.addons.TestUtils.renderIntoDocument(
+      <SampleAnchorClass anchorManager={manager} {...props}/>
+    );
+    anchors.push(anchor);
+    return anchor;
+  };
+
+  var unmountAnchor = function(anchor) {
+    React.unmountComponentAtNode(anchor.getDOMNode().parentNode);
+    anchors.splice(anchors.indexOf(anchor, 1));
+  };
+
+  beforeEach(function() {
+    clock = sinon.useFakeTimers();
+    manager = StubAnchorManager();
+    anchors = [];
+  });
+
+  afterEach(function() {
+    anchors.slice().forEach(unmountAnchor);
```

Comment:
Oh, so that's because we're iterating over an array while deleting items from it. Technically it *might* work depending on how `forEach` is implemented and/or how we're iterating over it, but I wanted to make sure that we e.g. don't skip any items, so I added the `slice` call.